src/blob_gc_picker_test.cc

```diff
@@ -74,24 +83,50 @@ TEST_F(BlobGCPickerTest, BeingGC) {
   AddBlobFile(2U, 1U, 0U);
   UpdateBlobStorage();
   blob_gc = basic_blob_gc_picker_->PickBlobGC(blob_storage_.get());
-  ASSERT_EQ(blob_gc->inputs().size(), 1);
-  ASSERT_EQ(blob_gc->inputs()[0]->file_number(), 2U);
+  ASSERT_EQ(blob_gc->gc_inputs().size(), 1);
+  ASSERT_EQ(blob_gc->gc_inputs()[0]->file_number(), 2U);
+  ASSERT_EQ(blob_gc->fs_inputs().size(), 1);
+  ASSERT_EQ(blob_gc->fs_inputs()[0]->file_number(), 2U);
 }
 
 TEST_F(BlobGCPickerTest, TriggerNext) {
   TitanDBOptions titan_db_options;
   TitanCFOptions titan_cf_options;
   titan_cf_options.max_gc_batch_size = 1 << 30;
+  titan_cf_options.max_fs_batch_size = 1 << 30;
+  titan_cf_options.min_gc_batch_size = 512 << 20;
+  titan_cf_options.min_fs_batch_size = 512 << 20;
   titan_cf_options.blob_file_target_size = 256 << 20;
+
+  // trigger next GC by gc job
+  titan_cf_options.merge_small_file_threshold = 30U << 20;
+  titan_cf_options.free_space_threshold = 1U << 30;
   NewBlobStorageAndPicker(titan_db_options, titan_cf_options);
   AddBlobFile(1U, 1U << 30, 1000U << 20);  // valid_size = 24MB
-  AddBlobFile(2U, 1U << 30, 512U << 20);   // valid_size = 512MB
-  AddBlobFile(3U, 1U << 30, 512U << 20);   // valid_size = 512MB
-  AddBlobFile(4U, 1U << 30, 512U << 20);   // valid_size = 512MB
+  AddBlobFile(2U, 1U << 30, 994U << 20);   // valid_size = 30MB
+  AddBlobFile(3U, 1U << 30, 994U << 20);   // valid_size = 30MB
+  AddBlobFile(4U, 1U << 30, 994U << 20);   // valid_size = 30MB
+  UpdateBlobStorage();
+  auto blob_gc = basic_blob_gc_picker_->PickBlobGC(blob_storage_.get());
+  ASSERT_TRUE(blob_gc != nullptr);
+  ASSERT_EQ(blob_gc->trigger_next(), true);
+
+  // trigger next GC by fs job
+  titan_cf_options.merge_small_file_threshold = 0U << 20;
+  titan_cf_options.free_space_threshold = 512U << 20;
+  NewBlobStorageAndPicker(titan_db_options, titan_cf_options);
+  AddBlobFile(1U, 1U << 30, 1000U << 20);  // discardable_size = 1000MB
+  AddBlobFile(2U, 1U << 30, 512U << 20);   // discardable_size = 512MB
+  AddBlobFile(3U, 1U << 30, 512U << 20);   // discardable_size = 512MB
+  AddBlobFile(4U, 1U << 30, 512U << 20);   // discardable_size = 512MB
   UpdateBlobStorage();
   auto blob_gc = basic_blob_gc_picker_->PickBlobGC(blob_storage_.get());
```

Comment:
conflicting declaration with the 110 line.

Reply:
fixed

Reply:
Good